cpg-analysis/src/main/kotlin/de/fraunhofer/aisec/cpg/query/FlowQueries.kt

```diff
@@ -406,6 +409,41 @@ fun Node.alwaysFlowsTo(
     vararg sensitivities: AnalysisSensitivity =
         ContextSensitive + FieldSensitive + FilterUnreachableEOG,
     predicate: (Node) -> Boolean,
+): QueryTree<Boolean> {
+    return this.alwaysFlowsToPrivate(
+        allowOverwritingValue = allowOverwritingValue,
+        noSinkIsGood = false,
+        earlyTermination = earlyTermination,
+        identifyCopies = identifyCopies,
+        stopIfImpossible = stopIfImpossible,
+        scope = scope,
+        sensitivities = sensitivities,
+        predicate = predicate,
+    )
+}
+
+/**
+ * This function tracks if the data in [this] always flow through a node which fulfills [predicate].
+ * An early termination can be specified by the predicate [earlyTermination].
+ * [allowOverwritingValue] can be used to configure if overwriting the value (or part of it) results
+ * in a failure of the requirement (if `false`) or if it does not affect the evaluation. The
+ * analysis can be configured with [scope] and [sensitivities]. [stopIfImpossible] enables the
+ * option to stop iterating through the EOG if we already left the function where we started and
+ * none of the nodes reaching by the DFG is in the new scope or one of its parents (i.e., the
+ * condition cannot be fulfilled anymore).
+ *
+ * If [noSinkIsGood] is set to `true`, all results with "ended path" are considered as fulfilled.
+ */
+private fun Node.alwaysFlowsToPrivate(
```

Comment:
IMO we should name this Internal and not Private because I think to remember that we did so in other places and that would be more consistent. But the author is allowed to merge if he does not agree.